### PR TITLE
converted Mojo::DOM tests to using subtests

### PR DESCRIPTION
**Pull request by rabbiveesh**
### Summary
I converted the tests for Mojo::DOM to using subtests.

### Motivation
I saw on IRC that we're planning on converting all tests to using subtests, and I wanted to give back a little (you guys literally make my livelihood).

### References
This addresses issue #1520. I hope to be able to do more in the future, too.

### CODE REVIEW

**Comment by kraih** [comment]
This PR needs to be rebased on master.

File: t/mojo/dom.t [hunk 1]
```diff
@@ -4,94 +4,101 @@ use Test::More;
...
+subtest 'Empty' => sub {
+  is(Mojo::DOM->new,                     '',    'right result');
+  is(Mojo::DOM->new(''),                 '',    'right result');
+  is(Mojo::DOM->new->parse(''),          '',    'right result');
+  is(Mojo::DOM->new->at('p'),            undef, 'no result');
+  is(Mojo::DOM->new->append_content(''), '',    'right result');
+  is(Mojo::DOM->new->all_text,           '',    'right result');
+};
+
+
```
Comment (kraih) [inline t/mojo/dom.t:16]: Extra newline.

**Comment by marcusramberg** [comment]
Looks good to me once it has been tidyed.

**Comment by rabbiveesh** [comment]
okay, i tidied it. Was I not supposed to forcepush it?

**Pull request merged**